### PR TITLE
fix TSGatherForward transformation

### PR DESCRIPTION
**Pull request by evkotov**
### Details:
 - fix TSGatherForward transformation
 - add unit tests

### Tickets:
 - 111270

### CODE REVIEW

File: src/common/transformations/src/transformations/transpose_sinking/ts_gather.cpp [hunk 1]
```diff
@@ -67,20 +67,62 @@ TSGatherForward::TSGatherForward() {
...
                 new_transpose_order[i] = order_val[j];
-                j++;
+                ++j;
+            } else if (order_val[j] == axis && static_cast<int>(k) <= n_axis_dims) {
+                // new dimensions within axis
+                new_transpose_order[i] = order_val[j] + k;
+                ++k;
+            } else {  // order_val[j] > axis
+                // dimensions after axis
+                new_transpose_order[i] = order_val[j] + indices_rank_val - 1;
```
Comment (itikhono) [inline src/common/transformations/src/transformations/transpose_sinking/ts_gather.cpp:122]: new_transpose_order[i] = order_val[j] + n_axis_dims;

**Review by itikhono — APPROVED** [review]

File: src/common/transformations/tests/transpose_sinking/ts_gather_test.cpp [hunk 2]
```diff
@@ -64,6 +64,7 @@ auto wrapper = [](const TestCase& test_case) {
 struct GatherForwardArguments {
     OutputVector inputs_to_main;
     Output<Node> new_input_to_Gather_1;
+    ov::AxisVector new_transpose_order;
```
Comment (t-jankowski) [inline src/common/transformations/tests/transpose_sinking/ts_gather_test.cpp:67]: Avoid using `ov::`, there's `using namespace ov;` at the beginning of the file (applies elsewhere)

File: src/common/transformations/tests/transpose_sinking/ts_gather_test.cpp [hunk 3]
```diff
@@ -80,9 +81,11 @@ auto test_forward_gather = [](const GatherForwardArguments& test_arguments) {
     test_case.model.model_template = create_model;
 
     // Reference model description:
-    auto new_transpose = [](const vector<size_t>& idxs, const OutputVector& out_vec) -> OutputVector {
+    auto new_transpose = [&test_arguments](const vector<size_t>& idxs, const OutputVector& out_vec) -> OutputVector {
         OutputVector new_out_vec(out_vec.size());
-        auto order = make_shared<Constant>(element::i32, Shape{4}, std::vector<int64_t>{3, 2, 1, 0});
+        auto order = make_shared<Constant>(element::i32,
```
Comment (t-jankowski) [inline src/common/transformations/tests/transpose_sinking/ts_gather_test.cpp:86]: Avoid using `element::` - `using namespace ov::element;` does the job.

File: src/common/transformations/src/transformations/transpose_sinking/ts_gather.cpp [hunk 4]
```diff
@@ -67,20 +67,62 @@ TSGatherForward::TSGatherForward() {
...
+                {1, 2, 3}      |        {7,5}         |   1  |      {1, 7, 5, 3}
+
+            New transpose order length equals to output Gather shape size.
+            As gather modifies input shape within axis dimension, our transpose order
+            will be modified with axis dimension.
+            New transpose order values:
+                - values before axis will be original
+                - values in [axis, axis + indexes_ranks_size - 1] will be original + [0 1 ...]
+                  if indexes_ranks_size == 0, there will be no such items
+                - values after axis with be original + indexes_rank_size - 1
```
Comment (t-jankowski) [inline src/common/transformations/src/transformations/transpose_sinking/ts_gather.cpp:86]:
```suggestion
                - values after axis will be original + indexes_rank_size - 1
```

**Review by t-jankowski — APPROVED** [review]

**Comment by jane-intel** [comment]
Just an observation. Why can't we reuse Squeeze / Unsqueeze TS code?

**Pull request merged**